uint8 TYPE_SHAKE=0
uint8 TYPE_SHAKE_5S=1
uint8 TYPE_TURN=2
uint8 type
